test/e2e/framework/docker.go

```diff
@@ -76,7 +78,30 @@ func (d *Docker) RunCommand(command ...string) (string, string) {
 	cmd.Stderr = &stderr
 
 	err := cmd.Run()
+
+	return stdout.String(), stderr.String(), err
+}
+
+func (d *Docker) RunCommand(command ...string) (string, string) {
+	stdout, stderr, err := d.runCommand(command...)
 	Expect(err).NotTo(HaveOccurred())
 
-	return stdout.String(), stderr.String()
+	return stdout, stderr
+}
+
+func (d *Docker) RunCommandUntil(command ...string) (string, string) {
+	var cmdErr error
+	var stdout, stderr string
+
+	err := wait.PollImmediate(5*time.Second, time.Duration(TestContext.OperationTimeout)*time.Second,
+		func() (bool, error) {
+			stdout, stderr, cmdErr = d.runCommand(command...)
+			if cmdErr != nil {
+				return false, nil // nolint:nilerr // Returning nil value is intentional
+			}
```

Comment:
I think we still want to report the command error if the retry timeout is reached. You could use `framework.AwaitUntil` or keep track of the last error and report it below.

Reply:
OK. My only nit is that it's not known if it's a transient error so perhaps change the output to "_Error attempting to run..._"

Reply:
I agree. Fixed and squashed the commits.